Modules/Quotas/src/Quotas.Domain/Aggregates/Tiers/Tier.cs

```diff
@@ -28,4 +28,16 @@ public Tier(TierId id, string name)
 
         return Result.Success<TierQuotaDefinition, DomainError>(quotaDefinition);
     }
+
+    public Result<TierQuotaDefinitionId, DomainError> DeleteQuota(string tierQuotaDefinitionId)
+    {
+        var quotaDefinition = Quotas.FirstOrDefault(q => q.Id == tierQuotaDefinitionId);
+
+        if (quotaDefinition == null)
+            return Result.Failure<TierQuotaDefinitionId, DomainError>(GenericDomainErrors.NotFound(nameof(TierQuotaDefinition)));
+
+        Quotas.Remove(quotaDefinition);
+
+        return Result.Success<TierQuotaDefinitionId, DomainError>(quotaDefinition.Id);
+    }
```

Comment:
Always remember that domain projects are the most important. So every method in such a project must be thoroughly unit tested.